src/internal/packager2/layout/import.go

```diff
@@ -46,10 +47,12 @@ func resolveImports(ctx context.Context, pkg v1alpha1.ZarfPackage, packagePath,
 		var importedPkg v1alpha1.ZarfPackage
 		if component.Import.Path != "" {
 			importPath := filepath.Join(packagePath, component.Import.Path)
-			if _, ok := seenImports[importPath]; ok {
-				return v1alpha1.ZarfPackage{}, fmt.Errorf("package %s imported in cycle by %s", filepath.ToSlash(importPath), filepath.ToSlash(packagePath))
+			importKey := fmt.Sprintf("%s-%s", component.Name, importPath)
+			if _, ok := seenImports[importKey]; ok {
+				return v1alpha1.ZarfPackage{}, fmt.Errorf("package %s imported in cycle by %s in component %s", filepath.ToSlash(importPath), filepath.ToSlash(packagePath), component.Name)
 			}
-			seenImports[importPath] = nil
+			seenImports = maps.Clone(seenImports)
```

Comment:
Ah so by using the maps.clone it creates a shallow copy so when the next component in the for loop comes by it doesn't have the sibling branches on it. Very neat!